Assets/Scripts/EOSAchievementManager.cs

```diff
@@ -275,22 +184,61 @@ private static AchievementsInterface GetEOSAchievementInterface()
         /// Invoked when the request from the server has completed - whether
         /// successful or not.
         /// </param>
-        private static void QueryAchievements(ProductUserId productUserId, OnQueryDefinitionsCompleteCallback callback)
+        private Task<List<DefinitionV2>> QueryAchievementsAsync(ProductUserId productUserId)
         {
             var options = new QueryDefinitionsOptions
             {
                 LocalUserId = productUserId
             };
 
+            TaskCompletionSource<List<DefinitionV2>> tcs = new();
+
             GetEOSAchievementInterface().QueryDefinitions(ref options, null, (ref OnQueryDefinitionsCompleteCallbackInfo data) =>
             {
                 if (data.ResultCode != Result.Success)
                 {
                     Log($"Unable to query achievement definitions. Result code: {data.ResultCode}");
                 }
 
-                callback?.Invoke(ref data);
+                tcs.SetResult(GetCachedAchievements());
             });
+
+            return tcs.Task;
+        }
+
+        /// <summary>
+        /// Retrieves the list of achievement definitions for the game that have
+        /// been cached locally.
+        /// </summary>
+        /// <returns>A list of achievement definitions.</returns>
+        private List<DefinitionV2> GetCachedAchievements()
```

Comment:
not useful criticism: It somewhat surprises me that I could call a method called "GetCachedAchievements", but it has to rebuild the list each time. I really wish we could know if this work needed to be re-done after its first calling. Thing is, I don't think Epic has provided a way of knowing that the provided achievements list is a different version than the past call...

Reply:
Yes - that is a valid critique of the EOS SDK in my opinion. That the SDK itself caches the achievements though I believe is a by-product of anti-cheat implementation - not necessarily a way to prevent the application from mis-representing achievements, but as a way to encourage use of _their_ cache as the source of truth over whatever locally cached version one might have made oneself.